service_darwin.go

```diff
@@ -206,13 +206,22 @@ func (s *darwinLaunchdService) Install() error {
 }
 
 func (s *darwinLaunchdService) Uninstall() error {
-	s.Stop()
+	if err := s.Stop(); err != nil {
+		return err
+	}
 
 	confPath, err := s.getServiceFilePath()
 	if err != nil {
 		return err
 	}
-	return os.Remove(confPath)
+	err = os.Remove(confPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+	return nil
```

Comment:
We should probably move the error into an if statement. We could even simplify further by doing  something like 
```suggestion
	if err = os.Remove(confPath); err != nil && !os.IsNotExist(err) {
		return err
	}
	return nil
```